creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1194,35 +1195,49 @@ class SnapbackSM {
    * @notice this will issue sync to healthy secondary and update replica set away from unhealthy secondary
    */
   async aggregateReconfigAndPotentialSyncOps(nodeUsers, unhealthyPeers) {
+    // Parallelize calling this._aggregateOps on chunks of 500 nodeUsers at a time
+    const nodeUserBatches = _.chunk(
+      nodeUsers,
+      AGGREGATE_RECONFIG_AND_POTENTIAL_SYNC_OPS_BATCH_SIZE
+    )
+    const results = []
+    for (const nodeUserBatch of nodeUserBatches) {
+      const resultBatch = await Promise.allSettled(
+        nodeUserBatch.map((nodeUser) =>
+          this._aggregateOps(nodeUser, unhealthyPeers)
+        )
+      )
+      results.push(...resultBatch)
+    }
+
+    // Combine each batch's requiredUpdateReplicaSetOps and potentialSyncRequests
     let requiredUpdateReplicaSetOps = []
     let potentialSyncRequests = []
-
-    for (const nodeUser of nodeUsers) {
-      const { primarySpID, secondary1SpID, secondary2SpID } = nodeUser
-
-      // If these spIds are undefined, it means that the Discovery Node does not have the latest code to return
-      // these fields. Fallback to the original aggregation logic. If present, use the spIds to determine whether
-      // reconfigs/syncs are necessary.
-      let subsetReplicaOps, subsetSyncReqs
-      if (primarySpID && secondary1SpID && secondary2SpID) {
-        ;({
-          requiredUpdateReplicaSetOps: subsetReplicaOps,
-          potentialSyncRequests: subsetSyncReqs
-        } = await this._aggregateOpsWithQueriedSpIds(nodeUser, unhealthyPeers))
-      } else {
-        // TODO: remove this else case once all the Discovery Nodes have upgraded and return the necessary replica set spIds
-        ;({
-          requiredUpdateReplicaSetOps: subsetReplicaOps,
-          potentialSyncRequests: subsetSyncReqs
-        } = await this._aggregateOpsWithoutQueriedSpIds(
-          nodeUser,
-          unhealthyPeers
-        ))
+    for (const promiseResult of results) {
+      // Skip and log failed promises
+      const {
+        status: promiseStatus,
+        value: reconfigAndSyncOps,
+        reason: promiseError
+      } = promiseResult
+      if (promiseStatus !== 'fulfilled') {
+        logger.error(
+          `aggregateReconfigAndPotentialSyncOps() encountered unexpected failure: ${promiseError}`
```

Comment:
last thing -- will promiseError be a string? if it's an error object, I want to avoid printing that out cus it gets un-readable in this logger and ${} formatting. if it is a string, lets leave as is. else lets get that string

Reply:
fair point -- changed to log string or Error